riptide-httpclient/src/test/java/org/zalando/riptide/httpclient/BufferingApacheClientHttpRequestTest.java

```diff
@@ -0,0 +1,23 @@
+package org.zalando.riptide.httpclient;
+
+import org.apache.hc.client5.http.classic.methods.HttpUriRequest;
+import org.junit.jupiter.api.Test;
+
+import java.net.URISyntaxException;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class BufferingApacheClientHttpRequestTest {
+
+    @Test
+    void shouldThrowIllegalArgumentException() throws URISyntaxException {
+        final HttpUriRequest httpUriRequest = mock(HttpUriRequest.class);
+        when(httpUriRequest.getUri()).thenThrow(URISyntaxException.class);
+
+        final BufferingApacheClientHttpRequest request = new BufferingApacheClientHttpRequest(null, httpUriRequest);
+        assertThrows(IllegalArgumentException.class, request::getURI);
+    }
+
+}
```

Comment:
side note - please add new line at the end of the file

Reply:
Don't know why my IDE didn't add new line - I've configured it to do so:
![image](https://github.com/zalando/riptide/assets/3616518/c3a395e4-507b-410c-854e-3b05f1ececd7)

Anyway, used this trick to fix it:
https://unix.stackexchange.com/a/161853/500398

```bash
git ls-files -z | while IFS= read -rd '' f; do if file --mime-encoding "$f" | grep -qv binary; then tail -c1 < "$f" | read -r _ || echo >> "$f"; fi; done
```

Reply:
May I suggest to use [.editorconfig](https://editorconfig.org/) for which there are IDE plugins available. Good chance contributors already have it installed.

Same goes for [.pre-commit-config.yaml](https://pre-commit.com/hooks.html) which among other useful hooks (newline at the end of each file, etc.) also supports editorconfig.